### ISSUE TITLE
SslStream on Unix might starve thread pool under load

### ISSUE THREAD

**Issue by dv00d00** [microsoft/reverse-proxy#108]
I've recently encountered a thread pool starvation in Kestrel. @davidfowl helped me to debug the issue and it looks like SslStream was the bad player (https://github.com/dotnet/aspnetcore/issues/21183)

TLDR: on restart under load, every https request was waiting on [return task.GetAwaiter().GetResult();](https://github.com/dotnet/runtime/blob/dbc7fc6ec8d3812cef5eee5d0a9492e0c1fc37e1/src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs#L112)

Would it be possible to fix it before 1.0 release? Docker container would probably be #1 host for reverse proxy and you would have no reason to put another reverse proxy that can handle TLS in front of this one.

**Comment by karelz** [comment]
Duplicate of https://github.com/dotnet/aspnetcore/issues/21183
